Assets/Scripts/Movement/ContinuousMovement.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.XR.CoreUtils;
+using UnityEngine.XR;
+using UnityEngine.XR.Interaction.Toolkit;
+
+public class ContinuousMovement : MonoBehaviour {
+
+    public float speed = 1;
+    public XRNode inputSource;
+    public float gravity = -9.81f;
+    public LayerMask groundLayer;
+    public float additionalHeight = 0.2f;
+
+    private float fallingSpeed;
+    private XROrigin rig;
+    private Vector2 inputAxis;
+    private CharacterController character;
+
+    // Start is called before the first frame update
+    void Start() {
+        character = GetComponent<CharacterController>();
+        rig = GetComponentInParent<XROrigin>();
+    }
+
+    // Update is called once per frame
+    void Update() {
+        InputDevice device = InputDevices.GetDeviceAtXRNode(inputSource);
+        device.TryGetFeatureValue(CommonUsages.primary2DAxis, out inputAxis);
+    }
+
+    void FixedUpdate() {
+        CapsuleFollowHeadset();
+
+        Quaternion headYaw = Quaternion.Euler(0, rig.Camera.transform.eulerAngles.y, 0);
+        
+        Vector3 direction = headYaw * new Vector3(inputAxis.x, 0, inputAxis.y);
+        character.Move(direction * Time.fixedDeltaTime * speed);
+
+        // gravity
+        bool isGrounded = CheckIfGrounded();
```

Comment:
conventionally, boolean methods are named something like `isGrounded()` -- might be nice to do that to avoid verbosity